src/components/contentAvail/contentAvailSrv.js

```diff
@@ -1,160 +1,230 @@
-                    if(hasSubscription){
-                        return true;
-                    }else{
-                        return res;
-                    }
-                });
-            }
-
-            function _isExamPurchased(purchaseData,examId){
-                var examKeyProp = idToKeyInStorage(examId);
-                return !!(purchaseData.exam === PURCHASED_ALL  || purchaseData.exam[examKeyProp]);
-            }
-
-            function _isFreeContent(freeContentData,pathArr){
-                var fullPath = pathArr.join('.');
-                var isFreeGetter = $parse(fullPath);
-                return !!isFreeGetter(freeContentData);
-            }
-
-            this.hasSubscription = function(){
-                return getUserPurchaseData().then(function(purchaseData){
-                    return _hasSubscription(purchaseData.subscription);
-                });
+    angular.module('znk.infra.contentAvail').provider('ContentAvailSrv', [
+        function () {
+
+            var _specials;
+
+            this.setSpecials = function(specialsObj) {
+                _specials = specialsObj;
             };
 
-            this.isDailyAvail = function(dailyOrder){
-                if(!angular.isNumber(dailyOrder) || isNaN(dailyOrder)){
-                    return $q.reject('daily order should be a number');
+            this.$get = ['$q', '$parse', '$injector', 'InfraConfigSrv', function($q, $parse, $injector, InfraConfigSrv) {
+
+                var PURCHASED_ALL = 'all';
+
+                function getUserPurchaseData(){
+                    var StorageService = InfraConfigSrv.getStorageService();
+                    var purchaseDataPath = StorageService.variables.appUserSpacePath + '/purchase';
+                    var defValues = {
+                        daily: 0,
+                        exam: {},
+                        tutorial: {},
+                        subscription: {}
+                    };
+                    return StorageService.get(purchaseDataPath,defValues);
                 }
-                return _baseIsEntityAvail().then(function(res){
-                    if(res === true){
-                        return true;
-                    }
 
-                    var purchaseData = res[0];
-                    var freeContent = res[1];
+                function getFreeContentData(){
+                    var StorageService = InfraConfigSrv.getStorageService();
+                    var freeContentPath = 'freeContent';
+                    var defValues = {
+                        daily: 0,
+                        exam: {},
+                        tutorial: {},
+                        specials: {}
+                    };
+                    return StorageService.get(freeContentPath,defValues);
+                }
 
-                    if(freeContent.daily >= dailyOrder){
-                        return true;
-                    }
+                function getUserSpecialsData(){
```

Comment:
Why just not to set a function which return the specials map instead of iterating ?